crates/rolldown/src/utils/prepare_build_context.rs

```diff
@@ -342,7 +342,10 @@ pub fn prepare_build_context(
     debug: raw_options.debug.is_some(),
     optimization: normalize_optimization_option(raw_options.optimization, platform),
     top_level_var: raw_options.top_level_var.unwrap_or(false),
-    minify_internal_exports: raw_options.minify_internal_exports.unwrap_or(false),
+    minify_internal_exports: raw_options.minify_internal_exports.unwrap_or(
+      matches!(format, OutputFormat::Esm)
+        || matches!(raw_minify, RawMinifyOptions::Bool(true) | RawMinifyOptions::Object(_)),
```

Comment:
I'll suggest using normalized minify options here. If minify it turned on by other options, will would like they have a chain affections.

Like if we have mode option, set it to `production` will enable `minify`, which then enables `minify_internal_exports`.

---

Just a thought. We could wait for we actually have the requiremnt.